pcf/core/particle.py

```diff
@@ -44,6 +44,27 @@ class Particle(object, metaclass=MetaParticle):
     Type of particle
     """
 
+    # create logger
+    dirname = os.path.dirname(__file__)
+    dirname, x = os.path.split(dirname)
+    filename = os.path.join(dirname, 'logging/pcf.log')
+    logger = logging.getLogger(getpass.getuser())
+    logger.setLevel(logging.DEBUG)
+
+    # create console handler and set level to debug
+    ch = logging.FileHandler(filename)
+    ch.setLevel(logging.DEBUG)
+
+    # create formatter
+    formatter = jsonlogger.JsonFormatter(
+        '%(asctime)s - %(name)s - %(module)s - %(funcName)s - %(levelname)s - %(message)s')
```

Comment:
this doesnt cvt messages to json. just adds extra fields such as time, name, level.

Reply:
that's what LogRecord does. JsonFormatter is built on top of it to turn it into json